### PR TITLE
refactor(nms):  Upgrade react-testing-library

### LINKED ISSUE #13120: Upgrade react-testing-library
We first have to upgrade to version [v12.1.5](https://github.com/testing-library/react-testing-library/releases/tag/v12.1.5) which requires replacing all `wait`s with `waitFor`s or `findBy*`.
Upgrading to 13 will happen together with the react 18 upgrade. 

Also upgrade @testing-library/react-hooks

### PR DESCRIPTION
**Pull request by HannaFar**
## Summary

- Closes #13120 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: nms/app/views/alarms/components/alertmanager/Receivers/__tests__/Receivers-test.tsx [hunk 1]
```diff
@@ -75,16 +69,14 @@ describe('Receivers', () => {
       fireEvent.click(getAllByText('View')[0]);
     });
     // clicking View should open the dialog
-    await waitForElement(() => getByText(/View Receiver/i));
-    expect(getByText(/View Receiver/i)).toBeInTheDocument();
+    // await waitForElement(() => getByText(/View Receiver/i));
```
Comment (thmsschmitt) [inline nms/app/views/alarms/components/alertmanager/Receivers/__tests__/Receivers-test.tsx]: superfluous comment

File: nms/app/views/alarms/components/alertmanager/Receivers/__tests__/Receivers-test.tsx [hunk 2]
```diff
@@ -75,16 +69,14 @@ describe('Receivers', () => {
...
+    expect(await findByText('View Receiver')).toBeVisible();
 
     // clicking Close should close the dialog
     act(() => {
       fireEvent.click(getByText(/close/i));
     });
-    await wait(() => {
-      expect(queryByText(/View Receiver/i)).not.toBeInTheDocument();
-    });
+    expect(await findByText('View Receiver')).not.toBeVisible();
```
Comment (thmsschmitt) [inline nms/app/views/alarms/components/alertmanager/Receivers/__tests__/Receivers-test.tsx]: I'm not sure about that on. `not.toBeVisible` is something else than `not.toBeInTheDocument`.

 I think we want something like
```
    await waitFor(() => {
      expect(queryByText('View Receiver')).not.toBeInTheDocument();
    });
```

File: nms/app/views/equipment/__tests__/EnodebConfigTest.tsx [hunk 3]
```diff
@@ -241,26 +240,23 @@ describe('<AddEditEnodeButton />', () => {
...
 
   it('Verify Enode Add', async () => {
-    const {getByTestId, getByText, queryByTestId} = render(<AddWrapper />);
-    await wait();
+    const {findByTestId, getByTestId, getByText, queryByTestId} = render(
+      <AddWrapper />,
+    );
     expect(queryByTestId('editDialog')).toBeNull();
-
     fireEvent.click(getByText('Add Enodeb'));
```
Comment (thmsschmitt) [inline nms/app/views/equipment/__tests__/EnodebConfigTest.tsx]: Interesting that we don't need `await findByTestId('editDialog')` here and in the other tests below given that the `wait` above was removed. Well if it works it works.

File: nms/app/views/equipment/__tests__/EquipmentEnodebTest.tsx [hunk 4]
```diff
@@ -177,7 +177,5 @@ describe('<Enodeb />', () => {
         new Date(currTime).toLocaleDateString(),
       );
     });
-    // TODO: The wait was needed as this test seems to be blinking.
```
Comment (thmsschmitt) [inline nms/app/views/equipment/__tests__/EquipmentEnodebTest.tsx:180]: 🤞

File: nms/app/views/subscriber/__tests__/SubscriberChartTest.tsx [hunk 5]
```diff
@@ -135,9 +135,9 @@ describe('<SubscriberChart />', () => {
...
   it('Verify Subscribers Data KPI', async () => {
-    const {getByTestId} = render(<Wrapper />);
-    await wait();
-    expect(getByTestId('Hourly Usage MB/s')).toHaveTextContent('12.11');
+    const {getByTestId, findByTestId} = render(<Wrapper />);
+
+    expect(await findByTestId('Hourly Usage MB/s')).toHaveTextContent('12.11');
     expect(getByTestId('Daily Avg MB/s')).toHaveTextContent('2.21');
     expect(getByTestId('Monthly Avg Mb/s')).toHaveTextContent('0.05');
     expect(getByTestId('Yearly Avg Mb/s')).toHaveTextContent('0.00');
```
Comment (Neudrino) [inline nms/app/views/subscriber/__tests__/SubscriberChartTest.tsx:143]: Should really only the first one be awaited?

File: nms/app/views/equipment/__tests__/EquipmentGatewayPoolTest.tsx [hunk 6]
```diff
@@ -351,33 +354,33 @@ describe('<AddEditGatewayPoolButton />', () => {
...
-    lteNetworkIdGatewayPoolsGatewayPoolIdGet.mockResolvedValue({
-      data: {
-        config: {mme_group_id: 4},
-        gateway_ids: [],
-        gateway_pool_id: 'pool4',
-        gateway_pool_name: 'pool4',
-      },
-    });
+    await waitFor(() =>
+      lteNetworkIdGatewayPoolsGatewayPoolIdGet.mockResolvedValue({
```
Comment (thmsschmitt) [inline nms/app/views/equipment/__tests__/EquipmentGatewayPoolTest.tsx:374]: It doesn't really make sense to wait for `mockResolvedValue` this is synchronous

File: nms/app/views/subscriber/__tests__/SubscriberAddEditTest.tsx [hunk 7]
```diff
@@ -451,29 +445,31 @@ describe('<AddSubscriberButton />', () => {
...
+          name: 'test_subscriber',
+          static_ips: {apn_0: '1.1.1.1'},
         },
-        name: 'test_subscriber',
-        static_ips: {apn_0: '1.1.1.1'},
-      },
+      });
     });
+
     // TODO: Test other tabs
```
Comment (Neudrino) [inline nms/app/views/subscriber/__tests__/SubscriberAddEditTest.tsx]: _Optional:_ I am surprised that this one survived. :wink:

File: nms/package.json [hunk 8]
```diff
@@ -84,8 +84,8 @@
     "@babel/preset-env": "^7.11.5",
     "@babel/preset-react": "^7.0.0",
     "@babel/preset-typescript": "^7.16.7",
-    "@testing-library/react": "^10.4.8",
-    "@testing-library/react-hooks": "^3.4.1",
+    "@testing-library/react": "^12.1.3",
```
Comment (thmsschmitt) [inline nms/package.json:87]: We can't upgrade to 13 because it requires React 18

**Review by thmsschmitt — APPROVED** [review]
I looked at @HannaFar's commits LGTM, I left some minor comments.

File: nms/app/views/equipment/__tests__/EquipmentGatewayPoolTest.tsx (hunk 6 above)
Comment (HannaFar) [reply nms/app/views/equipment/__tests__/EquipmentGatewayPoolTest.tsx:374]: That's right 😅

File: nms/app/views/equipment/__tests__/EnodebConfigTest.tsx (hunk 3 above)
Comment (HannaFar) [reply nms/app/views/equipment/__tests__/EnodebConfigTest.tsx]: I looked at the doc and it says that `queryBy` returns **null** if no match is found so there is no need to wait

**Review by Neudrino — APPROVED** [review]

**Pull request merged**